src/PlatformEngines.jl

```diff
@@ -592,16 +592,12 @@ end
 is_secure_url(url::AbstractString) =
     occursin(r"^(https://|\w+://(127\.0\.0\.1|localhost)(:\d+)?($|/))"i, url)
 
-function get_server_dir(url::AbstractString)
-    server = pkg_server()
+function get_server_dir(url::AbstractString, server=pkg_server())
     server === nothing && return
     startswith(url, server) || return
-    m = match(r"^\w+://([^\\/]+)$", server)
```

Comment:
> This should only strip `/` not `\\`.

I was wondering why there's a `\\` here. Can you explain it a bit?

Reply:
Because having a `/` or a `\\` in the server name makes it an invalid directory name—on Windows systems, it would put the files in a subdirectory, which is not what we want. Also, `\\` is just a weird thing to have in a URL, better to just not allow it.